lib/appsignal/finch.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Appsignal.Finch do
+  require Appsignal.Utils
+
+  @tracer Appsignal.Utils.compile_env(:appsignal, :appsignal_tracer, Appsignal.Tracer)
+  @span Appsignal.Utils.compile_env(:appsignal, :appsignal_span, Appsignal.Span)
+  import Appsignal.Utils, only: [module_name: 1]
+
+  @moduledoc false
+
+  require Logger
+
+  def attach do
+    handlers = %{
+      [:finch, :request, :start] => &__MODULE__.finch_request_start/4,
+      [:finch, :request, :stop] => &__MODULE__.finch_request_stop/4
```

Comment:
[Under certain conditions](https://github.com/sneako/finch/blob/0d32ee7ee3226035beb898b1f817ef4f5afbe6a8/test/finch/telemetry_test.exs#L233-L241), Finch will report `[:finch, :request, :exception]` *instead* of `[:finch, :request, :stop]`.

The suggestion makes sure the span is closed when this happens. I'm not sure if it would be possible to let the user know that it was an error somehow.

```suggestion
      [:finch, :request, :stop] => &__MODULE__.finch_request_stop/4
      [:finch, :request, :exception] => &__MODULE__.finch_request_stop/4
```

Reply:
> I'm not sure if it would be possible to let the user know that it was an error somehow.

It doesn't raise an error in Elixir unless you match on the result, so we don't catch it like that. It does appear to send [a stacktrace](https://github.com/sneako/finch/blob/0d32ee7ee3226035beb898b1f817ef4f5afbe6a8/test/finch/telemetry_test.exs#L201-L205) we can use to record an error. Is it possible to record an error this way?